api/genericresource/parse.go

```diff
@@ -0,0 +1,48 @@
+package genericresource
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/docker/swarmkit/api"
+)
+
+func newParseError(format string, args ...interface{}) error {
+	return fmt.Errorf("could not parse GenericResource: "+format, args...)
+}
+
+// Parse parses the GenericResource resources given by the arguments
+func Parse(cmd string, resources *api.Resources) error {
+	resources.Generic = make([]*api.GenericResource, 0)
+	rs := &resources.Generic
```

Comment:
I don't think pointer is required here. Something like this would do:

```
var rs []*api.GenericResource
...
rs = append(rs, ...)

resources.Generic = rs
```